drivers/serial/uart_ns16550.c

```diff
@@ -463,10 +471,14 @@ static int uart_ns16550_init(struct device *dev)
 {
 	uart_ns16550_configure(dev, &DEV_DATA(dev)->uart_config);
 
+	k_spinlock_key_t key = k_spin_lock(&DEV_DATA(dev)->lock);
```

Comment:
That's no need to lock here. As `_init()` is called at boot before scheduler runs.

Reply:
great, thanks